client/components/woopay/save-user/checkout-page-save-user.js

```diff
@@ -222,6 +237,20 @@ const CheckoutPageSaveUser = ( { isBlocksCheckout } ) => {
 		setPhoneNumber( getPhoneFieldValue() );
 	}, [ getPhoneFieldValue, isWCPayWithNewTokenChosen ] );
 
+	// Update the phone number when the billing phone is updated.
+	useEffect( () => {
+		if ( isCustomerDataUpdating ) {
+			setTimeout( () => {
+				setPhoneNumber( getPhoneFieldValue() );
+			}, 0 );
+		}
+	}, [
+		billingAddressPhone,
+		setPhoneNumber,
+		getPhoneFieldValue,
+		isCustomerDataUpdating,
```

Comment:
Used this approach due to inconsistencies when using the blur event and autofill.